### PR TITLE
Make engine reconnection test pass every time

### PR DESCRIPTION
**Pull request by franknoirot**

### CODE REVIEW

File: src/components/Stream.tsx [hunk 1]
```diff
@@ -157,6 +157,7 @@ export const Stream = () => {
   useEffect(() => {
     setIsFirstRender(kclManager.isFirstRender)
     if (!kclManager.isFirstRender) videoRef.current?.play()
+    setIsFreezeFrame(!kclManager.isFirstRender)
```
Comment (lf94) [inline src/components/Stream.tsx:160]: Muh brain but hey if it works LGTM lol

**Review by lf94 — APPROVED** [review]

File: src/components/Stream.tsx (hunk 1 above)
Comment (franknoirot) [reply src/components/Stream.tsx:160]: omg lol let me remove line 158 my bad. Just leaving contradictions in the code.

Comment (franknoirot) [reply src/components/Stream.tsx:160]: Haha I think the idea is that a freeze frame is by definition an in-the-middle state, and cannot possibly be a first render. Down in the rendering logic I saw `{(!isNetworkOkay || isLoading || isFirstRender) && !isFreezeFrame && (` must be resolving to `false` for no loading spinner to appear, and then saw that `&& !isFreezeFrame` we must not be in a freeze frame to show a loading spinner. So resets were being treated as freeze frames by not resetting them together.

Comment (lf94) [reply src/components/Stream.tsx:160]: (I actually added freeze frame) yeah it's meant to just say "imagine the stream is paused"

**Pull request merged**